### DISCUSSION TITLE
Can i CSS the caret ?

### DISCUSSION THREAD

**Discussion by DungnvKaopiz** [MathKeyboardEngine/MathKeyboardEngine#3] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/85537564/203221126-b8c5634a-233f-4f1f-9681-2da655d4515a.png)
How can i change the black square caret to a normal looking caret ?

**Comment by SymboLinker** [comment, marked as answer]
Hi @DungnvKaopiz,

### A short answer:
```
latexConfiguration.activePlaceholderShape = '|';
```

### A longer answer:

By default the [LatexConfiguration](https://github.com/MathKeyboardEngine/MathKeyboardEngine/blob/main/src/LatexConfiguration.ts) uses a `\blacksquare` as `activePlaceholderShape`. You can change this to anything that your typesetting library supports. Both KaTeX and MathJax support the `|` character and this character looks most like the caret/cursor that is often used in texts. On most keyboards it is above the Enter key and shares the key with `\`.

I've [updated the examples](https://github.com/MathKeyboardEngine/MathKeyboardEngine/commit/9d34e45f28abd118f69bfa853295e064d9fce73c) to use both the `|` and `\blacksquare`, depending on its position, like this (which is executed each time just before display):
```
if (keyboardMemory.current instanceof mke.Placeholder && keyboardMemory.current.nodes.length == 0) {
  latexConfiguration.activePlaceholderShape = String.raw`\blacksquare`;
} else {
  latexConfiguration.activePlaceholderShape = '|';
}
```
Note that also absolute values are entered with the `|` key. Therefore, you might want to also set the `activePlaceholderColor` just once at startup:
```
latexConfiguration.activePlaceholderColor = 'green';
```

Kind regards,
SymboLinker

**Reply by DungnvKaopiz** [reply]
thanks for your detail and fast reply, really appreciate it
